Packs/Gem/Integrations/Gem/Gem.py

```diff
@@ -0,0 +1,1211 @@
+            raise DemistoException(f'Failed to get token. Error: {str(e)}')
+
+    def _get_token(self):
+        """
+        Retrieves the authentication token for the Gem integration.
+
+        If the token is not found in the integration context or if it is expired,
+        a new token is generated and returned.
+
+        Returns:
+            str: The authentication token.
+        """
+        ctx = get_integration_context()
+
+        if not ctx or not ctx.get('auth_token'):
+            # No token in integration context, probably first run
+            auth_token = self._generate_token()
+        else:
+            # Token exists, check if it's expired and generate a new one if needed
+            auth_token = ctx.get('auth_token')
+            decoded_jwt = jwt.decode(auth_token, options={"verify_signature": False})  # type: ignore
+
+            token_expiration = datetime.fromtimestamp(decoded_jwt['exp'])
+
+            if token_expiration < datetime.now():
+                auth_token = self._generate_token()
+
+        return auth_token
+
+    def http_request(self, method: str, url_suffix='', full_url=None, headers=None, json_data=None, params=None, auth=True):
+        """
+        Sends an HTTP request to the specified URL, adding the required headers and authentication token.
+
+        Args:
+            method (str): The HTTP method to use (e.g., GET, POST, PUT, DELETE).
+            url_suffix (str, optional): The URL suffix to append to the base URL. Defaults to ''.
+            full_url (str, optional): The full URL to send the request to. If provided, `url_suffix` will be ignored.
+            Defaults to None.
+            headers (dict, optional): Additional headers to include in the request. Defaults to None.
+            json_data (dict, optional): JSON data to include in the request body. Defaults to None.
+            params (dict, optional): Query parameters to include in the request URL. Defaults to None.
+            auth (bool, optional): Whether to include authentication headers. Defaults to True.
+
+        Returns:
+            dict: The response from the HTTP request.
+
+        Raises:
+            Exception: If the request fails.
+
+        """
+        if auth:
+            headers = headers or {}
+            headers['Authorization'] = f'Bearer {self._auth_token}'
+        try:
+            response = super()._http_request(
+                method=method,
+                url_suffix=url_suffix,
+                full_url=full_url,
+                headers=headers,
+                json_data=json_data,
+                params=params,
+                raise_on_status=True
+            )
+            demisto.debug(f"Got response: {response}")
+            return response
+        except DemistoException as e:
+            demisto.error(f"Failed to execute {method} request to {url_suffix}. Error: {str(e)}")
+            raise Exception(f"Failed to execute {method} request to {url_suffix}. Error: {str(e)}")
```

Comment:
you can use http_request from the CommonServerPython http_request and to add error handler which outputs the new formatted error message.

Reply:
I'm using this function to add the Authorization header to the request

Reply:
@liormgem  I still recommend using the 'http_request' function from CommonServerPython.
 It's supported within XSOAR, ensuring compatibility, and we actively maintain it, promptly adapting it to any changes that may arise.
 
 Additionally, I'm available to assist you in integrating it into your code if needed.

However, if you still prefer to use your own 'http_request', you can proceed with that choice and resolve this note.